### PR TITLE
Include OLM support files

### PR DESCRIPTION
**Pull request by davidfestal**
The purpose of this PR is to include OLM CSV and linked files (OperatorHub-related files) into the `che-operator` repository, and allow pushing preview OperatorHub packages into a dedicated Quay.io application for both the Openshift 4.x and the upstream Kubernetes K8S flavours:
- `eclipse-che-operator-openshift/eclipse-che-preview-openshift`
- `eclipse-che-operator-kubernetes/eclipse-che-preview-kubernetes`

Each of the 2 pushed packages contain 2 channels:
- a `stable` channel with releases or pre-releases,
- a `nightly` channel, with the nightly build of the operator that by default starts a `CheCluster` with the Keycloak and Che nightly docker images

This PR also provides a number of scripts to simplify the lifecycle of these OLM packages, as well as the overall operator release process:
- `./update-nightly-olm-files.sh` creates a new version in the `nightly` channel of both the `eclipse-che-preview-openshift` and `eclipse-che-preview-kubernetes` packages, while updating them to maintain the consistency with the basic operator deployment yaml files (operator.yaml, rile.yaml, etc...)
- `./release-operator-code.sh <release>` changes the default tag of both the `che-server` and `che-keycloak` docker images to a given release in the operator `go` code , and build the operator docker image with the given release set as the docker image tag.
- `./release-olm-files.sh <release>` creates a new version in the `stable` channel of both the `eclipse-che-preview-openshift` and `eclipse-che-preview-kubernetes` packages. It creates it from the last corresponding nightly version but changing the version names to be consistent with both the new release name passed to the script, and the last release of the `stable` channel
- `./push-olm-files-to-quay.sh` pushes the whole content of the `eclipse-che-preview-openshift` and `eclipse-che-preview-kubernetes` packages to the `eclipse-che-operator-openshift` and `eclipse-che-operator-kubernetes` Quay.io applications.
- `./prepare-community-operators-update.sh` prepares, for both `kubernetes` and `openshift` platform, the changes required to create a PR on the [community-operators](https://github.com/operator-framework/community-operators) repository, that will add the latest release of each local `stable` channel into the official OperatorHub.

### CODE REVIEW

File: deploy/cluster_role.yaml [hunk 1]
```diff
@@ -22,6 +22,8 @@ rules:
       - get
       - delete
       - list
+      - patch
+      - update
```
Comment (rhopp) [inline deploy/cluster_role.yaml]: These operations were added to support possible update of oauthclient? In which situation this can happen?

**Comment by rhopp** [comment]
@nickboldt Yep... I guess that could be nice. 
To give you access to that jenkins instance I guess we need to make a Che committer of you ;-)

**Comment by nickboldt** [comment]
> @nickboldt Yep... I guess that could be nice.
> To give you access to that jenkins instance I guess we need to make a Che committer of you ;-)

haha, turns out I've already got access to https://ci.eclipse.org/che/ and can create new jobs.

File: deploy/operator.yaml [hunk 2]
```diff
@@ -16,30 +16,32 @@ spec:
...
   template:
     metadata:
       labels:
-        name: che-operator
+        app: che-operator
     spec:
-      serviceAccountName: che-operator
       containers:
         - name: che-operator
-          image: quay.io/eclipse-che/che-operator:7.0.0-beta-5.0
```
Comment (l0rd) [inline deploy/operator.yaml]: shouldn't we put nightly here as tag and changing the value only when releasing (ideally in a release branch)?

Comment (nickboldt) [reply deploy/operator.yaml]: +1, great idea (as long as we remember to set it to a specific version after branching for release). :D

**Comment by davidfestal** [comment]
This PR is a first step in the implementation of issue https://github.com/eclipse/che/issues/13780

File: deploy/operator.yaml (hunk 2 above)
Comment (davidfestal) [reply deploy/operator.yaml]: Then we have to switch to the `Always` pull policy I assume.

File: deploy/cluster_role.yaml (hunk 1 above)
Comment (davidfestal) [reply deploy/cluster_role.yaml]: I encountered this requirement during some tests I ran, but cannot remember precisely which one. FTR the initial rights Eugene added were `*` ;-)

File: deploy/operator.yaml (hunk 2 above)
Comment (davidfestal) [reply deploy/operator.yaml]: Done

File: olm/eclipse-che-test-kubernetes/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+generated/
```
Comment (l0rd) [inline olm/eclipse-che-test-kubernetes/.gitignore]: new line is missing

Comment (davidfestal) [reply olm/eclipse-che-test-kubernetes/.gitignore]: fixed

**Comment by davidfestal** [comment]
@l0rd I added all the basic management / release scripts, and included the changes you requested related to package and channel naming.
Do you want to review it once again before merging ?

**Pull request merged**